packages/thirdweb/src/wallets/in-app/core/authentication/linkProfile.ts

```diff
@@ -0,0 +1,115 @@
+import type { ThirdwebClient } from "../../../../client/client.js";
+import { getThirdwebBaseUrl } from "../../../../utils/domains.js";
+import type { Ecosystem } from "../../web/types.js";
+import type { Profile } from "./types.js";
+
+/**
+ * @description
+ * Links a new account to the current one using an auth token.
+ * For the public-facing API, use `wallet.addAuthMethod` instead.
+ *
+ * @internal
+ */
+export async function linkAccount({
+  client,
+  token,
```

Comment:
nit: would _prefer_ a more clear name for this, "token" is very generic, which token is this the "to link" or the "linking"